src/protocol/basics/reshare.rs

```diff
@@ -156,6 +181,45 @@ impl<'a, F: Field> Reshare<MaliciousContext<'a, F>, RecordId> for MaliciousRepli
     }
 }
 
+#[async_trait]
+/// Reshare algorithm for xor secret shares.
+impl<'a, B: Fp2Array> Reshare<SemiHonestContext<'a>, RecordId> for XorReplicated<B> {
+    async fn reshare<'fut>(
+        &self,
+        ctx: SemiHonestContext<'a>,
+        record_id: RecordId,
+        to_helper: Role,
+    ) -> Result<Self, Error>
+    where
+        SemiHonestContext<'a>: 'fut,
+    {
+        let r = ctx.prss().generate_bit_arrays(record_id);
+        let (s0, s1) = semi_honest_reshare(ctx, record_id, to_helper, self, r).await?;
+        Ok(XorReplicated::new(s0, s1))
+    }
+}
+
+#[async_trait]
+/// Malicious version of xor resharing is executed in semi-honest context.
+///
+/// As of March 2023, we aren't sure what is a proper way of doing reshare in malicious context.
+/// We believe that, even though an additive attack is possible, a malicious helper can ONLY
+/// corrupt the protocol output, but cannot learn any private info.
```

Comment:
I suggest you file an issue about this and link to it from here

Reply:
Opened #518